cudapoa/src/cudapoa_nw_banded.cu

```diff
@@ -211,29 +215,29 @@ __device__
         printf("graph %d, read %d\n", graph_count, read_length);
 #endif
 
-        for (IndexT graph_pos = 0; graph_pos < graph_count; graph_pos++)
+        for (SizeT graph_pos = 0; graph_pos < graph_count; graph_pos++)
         {
 
-            set_score(scores, 0, 0, (ScoreT)0, gradient, band_width, max_column);
+            set_score(scores, (SizeT)0, (SizeT)0, (ScoreT)0, gradient, band_width, max_column);
 
             SizeT node_id = graph[graph_pos];
-            IndexT i      = graph_pos + 1;
+            SizeT i       = graph_pos + 1;
 
             uint16_t pred_count = incoming_edge_count[node_id];
             if (pred_count == 0)
             {
-                set_score(scores, i, 0, gap_score, gradient, band_width, max_column);
+                set_score(scores, i, (SizeT)0, gap_score, gradient, band_width, max_column);
             }
             else
             {
-                ScoreT penalty = SHRT_MIN;
+                ScoreT penalty = score_type_min_limit;
                 for (uint16_t p = 0; p < pred_count; p++)
                 {
                     SizeT pred_node_id        = incoming_edges[node_id * CUDAPOA_MAX_NODE_EDGES + p];
                     SizeT pred_node_graph_pos = node_id_to_pos[pred_node_id] + 1;
-                    penalty                   = max(penalty, get_score(scores, pred_node_graph_pos, 0, gradient, band_width, read_length + 1, min_score_abs));
+                    penalty                   = max(penalty, get_score(scores, pred_node_graph_pos, (SizeT)0, gradient, band_width, static_cast<SizeT>(read_length + 1), min_score_value));
                 }
-                set_score(scores, i, 0, (ScoreT)(penalty + gap_score), gradient, band_width, max_column);
+                set_score(scores, i, (SizeT)0, static_cast<ScoreT>(penalty + gap_score), gradient, band_width, max_column);
```

Comment:
generally seeing lots of `(SizeT)` uses here (all preceding your changes). can you standardize those to static_cast :)?

Reply:
done. I was under impression that it is to define the type by compiler and no casting is involved. Regardless all such cases are now changed to `static_cast<>`

Reply:
Thanks @r-mafi ! I realized my earlier statement might have been confusing. All of those old type casts were "pre-dating" your changes, not preceding hehe. So none of this was from your changes, but I'm glad you were able to fix those in this PR too :)!